scripts/test-ssr/src/commands/main.ts

```diff
@@ -27,6 +28,36 @@ type MainParams = {
   stories: string;
 };
 
+function interceptConsoleLogs() {
+  const originalConsoleError = console.error;
+  const originalConsoleWarn = console.warn;
+
+  console.error = (...args: unknown[]) => {
+    // Ignoring 'aria-description' warning from React 17 as it's a valid prop
+    // https://github.com/facebook/react/issues/21035
+    if (containsAriaDescriptionWarning(args[0] + ' ' + args[1]) && React.version.startsWith('17')) {
+      return;
+    }
+
+    console.log(chalk.bgRed.whiteBright('❌  A console.error() was thrown during rendering HTML:'));
+    originalConsoleError(...args);
+
+    throw new Error('During rendering a console.error() was thrown');
+  };
+
+  console.warn = (...args: unknown[]) => {
```

Comment:
why do we need to hijack console.warn when there is no extra logic needed ?

Reply:
To ensure that `console.warn()` will not be called in tests

Reply:
how come console.warn throws an error ? is it some internal reacts logging logic ?

or we wanna throw on console warnings ?

Reply:
> or we wanna throw on console warnings ?

We want to throw on warnings, too. There should be no warnings in production mode in Node or in a browser.

Reply:
gotcha, can we document this intent as part of that function JSDoc ?

Reply:
Yup, done